File: examples/db_sqlite3.js
const { execFileSync } = require('child_process')
const fs = require('fs')

const db = 'sqlite3.db'
if (fs.existsSync(db)) {
  fs.unlinkSync(db)
}

const sql = [
  'CREATE TABLE persons (name TEXT);',
  "INSERT INTO persons VALUES ('alice'),('bob'),('charlie');",
  'SELECT rowid, name FROM persons;',
].join(' ')

const output = execFileSync('sqlite3', [db, sql], { encoding: 'utf8' })
process.stdout.write(output.replace(/\|/g, ' '))

File: examples/db_sqlite3.rb
require 'fileutils'
require 'open3'

db = 'sqlite3.db'
FileUtils.rm_f(db)

sql = [
  'CREATE TABLE persons (name TEXT);',
  "INSERT INTO persons VALUES ('alice'),('bob'),('charlie');",
  'SELECT rowid, name FROM persons;',
].join(' ')

stdout, stderr, status = Open3.capture3('sqlite3', db, sql)
abort(stderr) unless status.success?
print stdout.tr('|', ' ')
